selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/WorkspaceDtoDeserializer.java

```diff
@@ -38,26 +38,40 @@
 
   @Inject
   @Named("che.infrastructure")
-  private String infrastructure;
+  private Infrastructure infrastructure;
 
   public WorkspaceConfigDto deserializeWorkspaceTemplate(String templateName) {
     requireNonNull(templateName);
 
     try {
 
       URL url =
-          Resources.getResource(
-              WorkspaceDtoDeserializer.class,
-              format("/templates/workspace/%s/%s", infrastructure, templateName));
+          Resources.getResource(WorkspaceDtoDeserializer.class, getTemplateDirectory(templateName));
       return DtoFactory.getInstance()
           .createDtoFromJson(Resources.toString(url, Charsets.UTF_8), WorkspaceConfigDto.class);
     } catch (IOException | IllegalArgumentException | JsonSyntaxException e) {
       LOG.error(
           "Fail to read workspace template {} for infrastructure {} because {} ",
           templateName,
-          infrastructure,
+          getTemplateDirectory(templateName),
           e.getMessage());
       throw new RuntimeException(e.getLocalizedMessage(), e);
     }
   }
+
+  private String getTemplateDirectory(String template) {
+    String templateDirectoryName;
+    switch (infrastructure) {
+      case openshift:
+      case k8s:
+        templateDirectoryName = Infrastructure.openshift.toString().toLowerCase();
```

Comment:
Is it OK that case "k8s" but infrastructure which using is "openshift"?

Reply:
Also may be optimized like in the previous remark

Reply:
Yes - it was requirement to use _openshift_ templates for _k8s_.
About second suggestion - please take a look at the answer https://github.com/eclipse/che/pull/10384/files/a178ec69b067878bf86a824ead181e7232512185#r201723043